homeassistant/components/vallox/sensor.py

```diff
@@ -118,13 +126,21 @@ class ValloxSensor(SensorEntity):
     """Representation of a Vallox sensor."""
 
     def __init__(
-        self, name, state_proxy, metric_key, device_class, unit_of_measurement, icon
+        self,
+        name,
+        state_proxy,
+        metric_key,
+        device_class,
+        unit_of_measurement,
+        icon,
+        state_class=None,
     ) -> None:
         """Initialize the Vallox sensor."""
         self._name = name
         self._state_proxy = state_proxy
         self._metric_key = metric_key
         self._device_class = device_class
+        self._state_class = state_class
```

Comment:
Use `self._attr_state_class`  and you don't need the `state_class` property

Reply:
My idea was to switch all of them to their`_attr` equivalents in a follow up refactoring PR. The way it is now, the code stays coherent in style.

Reply:
Even better... this integration can probably use the `SensorEntityDescriptions` making most `_attr` and property methods obsolete 😉